File: csharp/PoS_after/PoS_after/PosTerminalTests/PosTerminalTests.cs
using NUnit.Framework;
using PosTerminal;

namespace PosTerminalTests
{
    public class Tests
    {
        [Test]
        public void VerifySale_AmountUnderSingleChargeLimit_IsApproved()
        {
            // This test uses the real CcVerifier class and takes a long time
            PosTerminal.PosTerminal sut = new PosTerminal.PosTerminal();
            
            Assert.DoesNotThrow(() =>
            {
                sut.VerifySale(10.0);
            });
        }

        [Test]
        [Ignore("Using a mock runs faster as we bypass the costly call to real CcVerifier")]
        public void VerifySale_viaMock_AmountUnderSingleChargeLimit_IsApproved()
        {
            ICcVerifier mockCcVerifier = new MockCcVerifier();

            PosTerminal.PosTerminal sut = new PosTerminal.PosTerminal(mockCcVerifier);
            
            Assert.DoesNotThrow(() =>
            {
                sut.VerifySale(10.0);
            });
        }
    }

    public class MockCcVerifier : ICcVerifier
    {
        public bool ApproveCharge(double amount)
        {
            return true;
        }
    }
}

File: java/PoS_after/src/test/java/POSTerminalTests.java
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.fail;

public class POSTerminalTests {

    @Test
    public void VerifySale_AmountUnderSingleChargeLimit_IsApproved() {
        // Using the real instance is costly in terms of time

        POSTerminal sut = new POSTerminal();

        try {
            sut.verifySale(10);
        } catch (POSTerminal.InvalidChargeException e) {
            fail("Amount was under Single Charge Amount - exception not expected!");
        }
    }

    @Test
    @Disabled("Remove to show faster feedback by using Test Double")
    public void VerifySale_ViaMock_AmountUnderSingleChargeLimit_IsApproved() {
        ICCVerifier mockCCVerifier = new MockCCVerifier();

        POSTerminal sut = new POSTerminal(mockCCVerifier);

        try {
            sut.verifySale(10);
        } catch (POSTerminal.InvalidChargeException e) {
            fail("Amount was under Single Charge Amount - exception not expected!");
        }
    }

    class MockCCVerifier implements ICCVerifier {

        @Override
        public boolean approveCharge(double amount) {
            return true;
        }
    }
}
